iOS Client/PodcastsController/PodcastFilterViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PodcastFilterViewController.swift
+//  iOS Client
+//
+//  Created by Jonathan Witten on 4/9/18.
+//  Copyright © 2018 Mockingbird. All rights reserved.
+//
+
+import Foundation
+import ReSwift
+import UIKit
+
+class PodcastsFilterViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, StoreSubscriber {
+    
+    @IBOutlet weak var tableView: UITableView!
+    var visibleStreams: Set<PodcastStream> = Set<PodcastStream>()
+    var streams: [PodcastStream] = [.pz, .mockingCast, .mockingPulpit]
```

Comment:
We are falling into another subtle trap here: two sources of truth.

This list exists here and in the state initialization (which I flagged). Perhaps this information should come from the state upon newState call or something

Reply:
So I don't think this is necessarily a violation of two sources of truth. We are deriving visibleStreams straight from the state in `newState` see `state.podcastsState.visiblePodcasts` and then using it as the data source for the table view. This is similar to storing the list of accounts straight from the state so when the table view reloads it has data to reference.

Reply:
Oh I think I see what you're saying. Derive the actual list of podcasts from state. Yeah that is something that I've been thinking about, we are hardcoding the list but in my opinion that's a pretty safe assumption because we aren't really deriving that list from anywhere except requirements from DZ which would take an app update to change anyways

Reply:
I think I'm going to leave this as a possible future refactor because it would take some rethinking on how we are managing the streams throughout the whole podcast flow if we were to derive the list of streams from a central data source (which honestly as I'm thinking more about it is probably the right way to go)